src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -168,28 +169,43 @@ public void phase1(final SnapshotIndexCommit snapshot) throws ElasticsearchExcep
             // Generate a "diff" of all the identical, different, and missing
             // segment files on the target node, using the existing files on
             // the source node
-            final Store.RecoveryDiff diff = recoverySourceMetadata.recoveryDiff(new Store.MetadataSnapshot(request.existingFiles()));
-            for (StoreFileMetaData md : diff.identical) {
-                response.phase1ExistingFileNames.add(md.name());
-                response.phase1ExistingFileSizes.add(md.length());
-                existingTotalSize += md.length();
-                if (logger.isTraceEnabled()) {
-                    logger.trace("[{}][{}] recovery [phase1] to {}: not recovering [{}], exists in local store and has checksum [{}], size [{}]",
-                            indexName, shardId, request.targetNode(), md.name(), md.checksum(), md.length());
+            final boolean recoverWithSyncId = recoverySourceMetadata.getCommitUserData().get(Engine.SYNC_COMMIT_ID) != null &&
```

Comment:
can you assign `recoverySourceMetadata.getCommitUserData().get(Engine.SYNC_COMMIT_ID)` to a local var makes it much easier to read